app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/NotificationsViewModel.kt

```
@@ -0,0 +1,142 @@
+                notificationsStore.notificationsFlow,
+                notificationsStore.settingsFlow
+            ) { notifications, settings ->
+                _state.value = _state.value.copy(
+                    loading = false,
+                    notifications = notifications,
+                    habits = habits,
+                    settings = settings,
+                    error = null
+                )
+            }.collect { } // Collect the combine flow
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)